flag.go

```diff
@@ -0,0 +1,212 @@
+// FlagSet to set a VMVar in the given VMVarMap. The VMVar sets a top-level arg
+// string literal in a jsonnet VM.
+//
+// The flag value on the command line is parsed as "key[=value]". If the value
+// is omitted then key is looked up in the environment for the value. It is an
+// error if the value is not provided and the environment variable does not
+// exist.
+//
+// The flag can be repeated multiple times on the command line. Each instance
+// adds a new value to the given VMVarMap.
+func TLAStrVar(fs *flag.FlagSet, m VMVarMap, name, usage string) {
+	fs.Var(vmVarMapValue{m, NewTLAStr, true}, name, usage)
+}
+
+// TLACodeVar defines flag with the given name and usage string in the given
+// FlagSet to set a VMVar in the given VMVarMap. The VMVar sets a top-level arg
+// code literal in a jsonnet VM.
+//
+// The flag value on the command line is parsed as "key[=value]". If the value
+// is omitted then key is looked up in the environment for the value. It is an
+// error if the value is not provided and the environment variable does not
+// exist.
+//
+// The flag can be repeated multiple times on the command line. Each instance
+// adds a new value to the given VMVarMap.
+func TLACodeVar(fs *flag.FlagSet, m VMVarMap, name, usage string) {
+	fs.Var(vmVarMapValue{m, NewTLACode, true}, name, usage)
+}
+
+// TLAStrFileVar defines flag with the given name and usage string in the given
+// FlagSet to set a VMVar in the given VMVarMap. The VMVar sets a top-level arg
+// string import in a jsonnet VM.
+//
+// The flag value on the command line is parsed as "key=file".
+//
+// The flag can be repeated multiple times on the command line. Each instance
+// adds a new value to the given VMVarMap.
+func TLAStrFileVar(fs *flag.FlagSet, m VMVarMap, name, usage string) {
+	fs.Var(vmVarMapValue{m, NewTLAStrFile, false}, name, usage)
+}
+
+// TLACodeFileVar defines flag with the given name and usage string in the
+// given FlagSet to set a VMVar in the given VMVarMap. The VMVar sets a
+// top-level arg code import in a jsonnet VM.
+//
+// The flag value on the command line is parsed as "key=file".
+//
+// The flag can be repeated multiple times on the command line. Each instance
+// adds a new value to the given VMVarMap.
+func TLACodeFileVar(fs *flag.FlagSet, m VMVarMap, name, usage string) {
+	fs.Var(vmVarMapValue{m, NewTLACodeFile, false}, name, usage)
+}
+
+// vmVarMapValue wraps a VMVarMap with the additional information it needs to
+// parse the various types of VMVars, using a makevar function to construct the
+// values in the map, and flagging whether a value can come from the evironment
+// or must be parsed from the command line.
+type vmVarMapValue struct {
+	m       VMVarMap
+	makevar func(string) VMVar
+	fromenv bool
+}
+
+// Set sets the flag var value from v. It implements the flag.Value interface.
+func (mv vmVarMapValue) Set(v string) error {
+	return mv.m.SetVar(v, mv.makevar, mv.fromenv)
```

Comment:
actually this is really neat how concisely you got to do what needs doing.

Reply:
I was quite pleased how that came together once I figured out the proper place in order to share the implementation for both kong and flags.